javalib/src/main/scala/java/lang/String.scala

```diff
@@ -667,22 +668,259 @@ final class _String()
 
-  def toUpperCase(locale: Locale): _String = toCase(Character.toUpperCase)
+  // Ported from Scala.js, commit: ac38a148, dated: 2020-09-25
+  def toUpperCase(locale: Locale): String = {
+    locale.getLanguage() match {
+      case "lt"        => toUpperCaseLithuanian()
+      case "tr" | "az" => toUpperCaseTurkishAndAzeri()
+      case _           => toUpperCase()
+    }
+  }
 
-  @inline
-  def toUpperCase(): _String = toCase(Character.toUpperCase)
+  private def toUpperCaseLithuanian(): String = {
+    /* Relevant excerpt from SpecialCasing.txt
+     *
+     * # Lithuanian
+     *
+     * # Lithuanian retains the dot in a lowercase i when followed by accents.
+     *
+     * # Remove DOT ABOVE after "i" with upper or titlecase
+     *
+     * 0307; 0307; ; ; lt After_Soft_Dotted; # COMBINING DOT ABOVE
+     */
+
+    /* Tests whether we are in an `After_Soft_Dotted` context.
+     * From Table 3.17 in the Unicode standard:
+     * - Description: There is a Soft_Dotted character before C, with no
+     *   intervening character of combining class 0 or 230 (Above).
+     * - Regex, before C: [\p{Soft_Dotted}]([^\p{ccc=230} \p{ccc=0}])*
+     *
+     * According to https://unicode.org/Public/13.0.0/ucd/PropList.txt, there
+     * are 44 code points with the Soft_Dotted property. However,
+     * experimentation on the JVM reveals that the JDK (8 and 14 were tested)
+     * only recognizes 8 code points when deciding whether to remove the 0x0307
+     * code points. The following script reproduces the list:
+
+for (cp <- 0 to Character.MAX_CODE_POINT) {
+  val input = new String(Array(cp, 0x0307, 0x0301), 0, 3)
+  val output = input.toUpperCase(new java.util.Locale("lt"))
+  if (!output.contains('\u0307'))
+    println(cp.toHexString)
+}
+
+     */
+    def afterSoftDotted(i: Int): scala.Boolean = {
+      val j = skipCharsWithCombiningClassOtherThanNoneOrAboveBackwards(i)
+      j > 0 && (codePointBefore(j) match {
+        case 0x0069 | 0x006a | 0x012f | 0x0268 | 0x0456 | 0x0458 | 0x1e2d |
+            0x1ecb =>
+          true
+        case _ => false
+      })
+    }
+
+    val preprocessed = replaceCharsAtIndex { i =>
+      (this.charAt(i): @switch) match {
+        case '\u0307' if afterSoftDotted(i) => ""
+        case _                              => null
+      }
+    }
+
+    preprocessed.toUpperCase()
+  }
+
+  private def toUpperCaseTurkishAndAzeri(): String = {
+    /* Relevant excerpt from SpecialCasing.txt
+     *
+     * # Turkish and Azeri
+     *
+     * # When uppercasing, i turns into a dotted capital I
+     *
+     * 0069; 0069; 0130; 0130; tr; # LATIN SMALL LETTER I
+     * 0069; 0069; 0130; 0130; az; # LATIN SMALL LETTER I
+     */
+
+    val preprocessed = replaceCharsAtIndex { i =>
+      (this.charAt(i): @switch) match {
+        case '\u0069' => "\u0130"
+        case _        => null
+      }
+    }
+
+    preprocessed.toUpperCase()
+  }
+  @inline def toUpperCase(): _String = toCase(Character.toUpperCase)
```

Comment:
Missing blank line. Also, to reduce the overall diff:
```suggestion
  }
  
  @inline
  def toUpperCase(): _String = toCase(Character.toUpperCase)
```